### ISSUE TITLE
Image upload for nested metric not working[App] 

### ISSUE THREAD

**Issue by AllaVinner** [wandb/client#2705]
**Describe the bug**
I want to log a list of images in a nested dictionary i.e. {'triain': {'images': image_list}}, however when I go to the app I get the message:
`'This run didn't log images for key 'train.image' step 0 index 0`
Note that the images do exist in the app folder: Files/media/images/image_list_x_y

**To Reproduce**
Code:
```
import wandb
import torch
wandb.login()
run = wandb.init(project= "MyProject", name= "wandb images")
im = torch.rand(3,40,40)
im_list = [wandb.Image(im), wandb.Image(im)]
im_dict = {'images': im_list}

run.log({'Image_list': im_list,
         'Image_dict': im_dict,
         })
run.finish()
```

In App:
1. Go to 'wandb images'
2. Scroll down to 'Media'
3. See error under 'Image_dict.Nested_im_list' but not under 'image_list'

**Expected behavior**
I expected to be able to see both sets of images.

**Screenshots**
![Screenshot_Wandb_Issue](https://user-images.githubusercontent.com/22410133/135256606-48b11a2b-ca4e-42f1-a2e7-35ddca052100.JPG)

**Operating System**
 - OS: Windows
 - Browser: Firefox
 - Version 0.12.2

**Notes**
First time I am posting so do let me know if I could have done something more clear or in a different way.

**Comment by vanpelt** [comment]
Hey @AllaVinner can you share a link to a run that's exhibiting this behavior?  If it's a private run only our support staff will be able to access it assuming you give us permission to investigate.

**Comment by AllaVinner** [comment]
Ofcourse, I link the toy-example which I described:
[Link](https://wandb.ai/joelottosson/MyProject?workspace)

**Comment by vanpelt** [comment]
Hey @AllaVinner this looks a bug in the client library, we likely won't be able to fix out for a couple weeks.  In the meantime, the best solution is to flatten the dict into individual keys, i.e.

```
wandb.log({"images_dict.images": [wandb.Image(...)]})
```

**Comment by AllaVinner** [comment]
Great! 

That works,

 thank you
